src/uploader.ts

```diff
@@ -26,34 +53,35 @@ export default class Uploader {
    *
    * @param {Function} onPreview - callback fired when preview is ready
    */
-  uploadSelectedFile({ onPreview }) {
-    const preparePreview = function (file) {
+  uploadSelectedFile({ onPreview }: UploadOptions) {
+    const preparePreview = function (file: File) {
       const reader = new FileReader();
 
       reader.readAsDataURL(file);
       reader.onload = (e) => {
-        onPreview(e.target.result);
+        onPreview((e.target as FileReader).result as string);
       };
     };
 
     /**
      * Custom uploading
      * or default uploading
      */
-    let upload;
+    let upload: Promise<UploadResponseFormat>;
 
     // custom uploading
     if (this.config.uploader && typeof this.config.uploader.uploadByFile === 'function') {
-      upload = ajax.selectFiles({ accept: this.config.types }).then((files) => {
+      const uploadByFile = this.config.uploader.uploadByFile;
+      upload = ajax.selectFiles({ accept: this.config.types || ''}).then((files: File[]) => {
```

Comment:
seems redundant

Reply:
`this.config` is a shared variable inside the class. I guess that the reason why the Typescript complains about undefined possibility inside an asynchronous block.